### ISSUE TITLE
-fpermissive errors

### ISSUE THREAD

**Issue by Kangz** [danielaparker/jsoncons#64]
Hey using jsoncons I get the following -fpermissive error. I could disable it locally, but ideally this could be fixed upstream.

Thanks!
```
jsoncons/src/jsoncons/json.hpp:2825:25: error: declaration of ‘typedef jsoncons::basic_json<StringT, Alloc>::member_type jsoncons::basic_json<StringT, Alloc>::name_value_pair’ [-fpermissive]
     typedef member_type name_value_pair;
                         ^
jsoncons/src/jsoncons/json_structures.hpp:250:7: error: changes meaning of ‘name_value_pair’ from ‘class jsoncons::name_value_pair<std::basic_string<typename StringT::value_type, std::char_traits<typename StringT::value_type>, typename std::allocator_traits<_Alloc>::rebind_alloc<typename StringT::value_type> >, jsoncons::basic_json<StringT, Alloc> >’ [-fpermissive]
 class name_value_pair
       ^
```

**Comment by danielaparker** [comment]
That was fixed on master, with fixes for g++ "4.9.2" contributed by massimomorara.

I've incorporated those fixes as a patch to the last release Release 0.98.4, in Release 0.98.4.1

**Comment by Kangz** [comment]
Great, thank you very much! I was using version 0.98.4, I'll upgrade.
